src/mesos_state_client.erl

```diff
@@ -51,13 +102,7 @@ poll(URI) ->
   {ok, Hostname} = inet:gethostname(),
   UserAgent = lists:flatten(io_lib:format("Mesos-State / Host: ~s, Pid: ~s", [Hostname, os:getpid()])),
   Headers = [{"Accept", "application/json"}, {"User-Agent", UserAgent}],
-  Headers1 =
-    case application:get_env(?APP, token) of
-      undefined ->
-        Headers;
-      {ok, Value} ->
-        [{"Authorization", Value}|Headers]
-    end,
+  Headers1 = maybe_use_token(Headers),
```

Comment:
So, I would probably rather add the code to load the token file (from environment, or if we know it's somewhere on disk in the mesos_state_app. Having it here seems a bit awkward.